### PR TITLE
pool incrementalHash so that we don't create one unnecessarily

### PR DESCRIPTION
**Pull request by heejaechang**
save cost of creating new incremental hash.

### CODE REVIEW

File: src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs [hunk 1]
```diff
@@ -14,41 +15,44 @@ namespace Microsoft.CodeAnalysis
     // all these are just helper methods
     internal partial class Checksum
     {
+        private static readonly ObjectPool<IncrementalHash> s_objectPool =
```
Comment (sharwell) [inline src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs]: 💡 Consider using a more specific name for this pool field

**Comment by heejaechang** [comment]
tagging @dotnet/roslyn-ide

File: src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs (hunk 1 above)
Comment (heejaechang) [reply src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs]: sure.

File: src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs [hunk 2]
```diff
@@ -14,41 +15,44 @@ namespace Microsoft.CodeAnalysis
     // all these are just helper methods
     internal partial class Checksum
     {
+        private static readonly ObjectPool<IncrementalHash> s_incrementalHashPool =
+            new ObjectPool<IncrementalHash>(() => IncrementalHash.CreateHash(HashAlgorithmName.SHA1), size: 20);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs]: why 20?

File: src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs [hunk 3]
```diff
@@ -14,41 +15,44 @@ namespace Microsoft.CodeAnalysis
...
-            using (var hash = IncrementalHash.CreateHash(HashAlgorithmName.SHA1))
+            using (var pooledHash = s_incrementalHashPool.GetPooledObject())
+            using (var pooledBuffer = SharedPools.ByteArray.GetPooledObject())
             {
-                using (var pooledBuffer = SharedPools.ByteArray.GetPooledObject())
-                {
-                    stream.Seek(0, SeekOrigin.Begin);
+                stream.Seek(0, SeekOrigin.Begin);
+
+                var hash = pooledHash.Object;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs]: does anything need to reset this, either before/after using it?

File: src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs (hunk 2 above)
Comment (heejaechang) [reply src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs]: no particular reason. other pool uses 20 so, it uses 20 as well.

File: src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs (hunk 3 above)
Comment (heejaechang) [reply src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs]: hash got reset when it get into the pool

**Pull request merged**